### ISSUE TITLE
`no-return-assign` could use an `except-parens` option to match `no-cond-assign`

### ISSUE THREAD

**Issue by moll** [eslint/eslint#2809]
Hey,

The `no-return-assign` could use an `except-parens` option to match `no-cond-assign` for those cases where returning and assign is convenient.

Thanks!

**Comment by nzakas** [comment]
Thanks for the note. Unfortunately, there's not enough information in this description. Can you please provide:

1. Some sample code you want to lint
2. The configuration of no-return-assign you're using
3. What ESLint currently outputs
4. What you want to happen instead

**Comment by moll** [comment]
http://eslint.org/docs/rules/no-cond-assign.html has:
> except-parens (default): Disallow assignments unless they are enclosed in parentheses.

I'm proposing `no-return-assign` do the same.

The following patterns would then not be considered warnings:
```javascript
function doSomething() {
    return (foo = bar + 2)
}

function doSomething() {
    return (foo += 2)
}
```

**Comment by nzakas** [comment]
Seems rational to me.

**Comment by mathieumg** [comment]
I know this is already accepted, just wanted to show further support for this.

**Comment by mysticatea** [comment]
I will try to make this.
